contracts/bonding/BondingManager.sol

```diff
@@ -498,41 +529,133 @@ contract BondingManager is IBondingManager, Controllable {
     }
 
+    function transcoderRates(address _transcoder) public constant returns (uint256, uint256, uint256, uint256, uint256, uint256) {
+        Transcoder storage t = transcoders[_transcoder];
+
+        return (t.rates[0], t.rates[1], t.rates[2], t.pendingRates[0], t.pendingRates[1], t.pendingRates[2]);
+    }
+
+    function transcoderStatus(address _transcoder) public constant returns (TranscoderStatus) {
+        Transcoder storage t = transcoders[_transcoder];
+
+        if (t.withdrawRound > 0) {
+            // Transcoder resigned
+            if (roundsManager().currentRound() >= t.withdrawRound) {
+                return TranscoderStatus.Unbonded;
+            } else {
+                return TranscoderStatus.Unbonding;
+            }
+        } else if (t.initialized) {
+            return TranscoderStatus.Registered;
+        } else {
+            return TranscoderStatus.NotRegistered;
+        }
+    }
+
     /*
-     * @dev Returns bonded stake for a delegator. Accounts for rewards since last state transition
+     * @dev Computes delegator status
      * @param _delegator Address of delegator
      */
-    function delegatorStake(address _delegator) public constant returns (uint256) {
-        // Check for valid delegator
-        if (!delegators[_delegator].initialized) throw;
+    function delegatorStatus(address _delegator) public constant returns (DelegatorStatus) {
+        Delegator storage del = delegators[_delegator];
+
+        if (del.withdrawRound > 0) {
+            // Delegator called unbond
+            if (roundsManager().currentRound() >= del.withdrawRound) {
+                return DelegatorStatus.Unbonded;
+            } else {
+                return DelegatorStatus.Unbonding;
+            }
+        } else if (del.transcoderAddress != address(0) && transcoders[del.transcoderAddress].withdrawRound > 0) {
+            // Transcoder resigned
+            if (roundsManager().currentRound() >= transcoders[del.transcoderAddress].withdrawRound) {
+                return DelegatorStatus.Unbonded;
+            } else {
+                return DelegatorStatus.Unbonding;
+            }
+        } else if (del.startRound > roundsManager().currentRound()) {
+            // Delegator round start is in the future
+            return DelegatorStatus.Pending;
+        } else if (del.startRound <= roundsManager().currentRound()) {
+            // Delegator round start is now or in the past
+            return DelegatorStatus.Bonded;
+        } else {
+            // Delegator in inactive phase
+            return DelegatorStatus.Inactive;
+        }
+    }
 
-        return delegators[_delegator].bondedAmount.add(delegatorRewards(_delegator));
+    /*
+     * @dev Return number of minted tokens for a reward call
+     */
+    function mintedTokensPerReward() public constant returns (uint256) {
+        return initialTokenSupply.mul(initialYearlyInflation).div(100).div(roundsManager().rewardCallsPerYear());
```

Comment:
I believe we said each active transcoder can call `reward` once per round. `roundsManager().rewardCallsPerYear()` just calculates the number of reward calls per year given the current round length and the protocol defined number of active transcoders